Add form for `TokenLightRuleElement`

In3luki opened the pull request:
![image](https://github.com/foundryvtt/pf2e/assets/41452412/4ec796a8-f9c3-496d-a1d4-9038ac0f6bbb)

![image](https://github.com/foundryvtt/pf2e/assets/41452412/322118a9-b40e-43e7-9650-f75a6807b2c7)

![image](https://github.com/foundryvtt/pf2e/assets/41452412/de9968d0-77f7-43c0-a8a6-4cddce844a0b)

![image](https://github.com/foundryvtt/pf2e/assets/41452412/5cffb6c7-9952-4904-8a90-d94ec80632d7)

Closes #8743

CarlosFdez commented on src/module/item/sheet/rule-elements/base.ts:
```diff
@@ -214,7 +253,12 @@ function cleanDataUsingSchema(schema: Record<string, DataField>, data: Record<st
+                cleanDataUsingSchema(field.fields as Record<string, DataField>, maybeObject);
+            } else {
+                cleanDataUsingSchema(field.fields as Record<string, DataField>, data);
+            }
```
Shouldn't all of this act on data[key]? This was supposed to be the recursive case, and I believe acting on data was probably incorrect.

In3luki replied:
I've added that check because it was passing non-objects in some instances.

CarlosFdez replied:
In those cases, what was the schema being passed?

In3luki replied:
The `text` field on a `RollNoteRuleElement` which is a `DataUnionField` that can be a string or an object (I think).

CarlosFdez commented on src/module/item/sheet/rule-elements/token-light-form.ts:
```diff
@@ -0,0 +1,60 @@
+    protected override tabNames = ["basic", "animation", "advanced"];
+    protected override tabDisplayStyle = "grid";
+
+    override async getData(): Promise<TokenLightSheetData> {
```
What happens if all of this is moved to getInitialValue(), and it gets converted from private to protected?

In3luki replied:
getInitialValue is called in every getData so I don't know if that is a good place. Maybe in the constructor?

CarlosFdez replied:
getData() is also called in every getData() though

In3luki replied:
Maybe I misunderstood what you meant. I thought you were talking about the tab declarations.

CarlosFdez replied:
Oh, sorry, I mean the default settings stuff

In3luki replied:
I'll look into it tomorrow.

In3luki commented:
Updated to apply suggestions:

- Made `getInitialValue` a protected method.
- Moved `LightData` construction for unowned items from `getData` to `getInitialValue`.
- Use initial value in template instead of duplicating it with a new object.
- Simplified the object check for recursive `cleanDataUsingSchema` calls.

In3luki commented:
Now that #10152 is merged, tab persistence works as expected.
